pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -664,21 +675,34 @@ func TestPrometheusCodec_DecodeEncodeLabelsQueryRequest(t *testing.T) {
 			expectedGetEndOrDefault:   1708588800 * 1e3,
 		},
 		{
-			name:        "zero limit is not allowed",
-			url:         "/api/v1/label/job/values?limit=0",
-			expectedErr: "limit parameter must be a positive number: 0",
+			name:        "zero limit is allowed",
+			url:         "/api/v1/label/job/values?limit=0&start=1708502400&end=1708588800",
+			expectedURL: "/api/v1/label/job/values?end=1708588800&start=1708502400", // Zero limit is omitted (it's the default).
+			expectedStruct: &PrometheusLabelValuesQueryRequest{
+				Path:      "/api/v1/label/job/values",
+				LabelName: "job",
+				Start:     1708502400 * 1e3,
+				End:       1708588800 * 1e3,
+				Limit:     0,
+			},
+			expectedGetLabelName:      "job",
+			expectedLimit:             0,
+			expectedGetStartOrDefault: 1708502400 * 1e3,
+			expectedGetEndOrDefault:   1708588800 * 1e3,
 		},
```

Comment:
looks like this fixes https://github.com/grafana/mimir/issues/11831 cc @itsmylife